src/compas/plugins.py

```diff
@@ -33,6 +35,17 @@
     "PluginDefaultNotAvailableError",
 ]
 
+if compas.RHINO:
+    import System  # type: ignore  # pragma: no cover
+
+    DotNetException = System.Exception  # pragma: no cover
+else:
+
+    class DummyDotNetException(BaseException):
+        pass
+
+    DotNetException = DummyDotNetException
```

Comment:
why is this done in two steps?

Reply:
what do you mean two steps? why I'm not using `BaseException` directly?

Reply:
no other reason except it never crossed my mind :)